frontend/src/lib/api.ts

```diff
@@ -16,3 +22,35 @@ export const loadData = async <T>(
   }
   return await response.json()
 }
+
+export const fetchAlgoliaData = async <T>(
+  indexName: string,
+  query = '',
+  currentPage = 0
+): Promise<AlgoliaResponseType<T>> => {
+  try {
+    const { results } = await client.search({
+      requests: [
+        {
+          indexName: `${NEST_ENV}_${indexName}`,
+          query,
+          hitsPerPage: 25,
+          page: currentPage - 1,
+          attributesToHighlight: [],
+        },
```

Comment:
We want to keep the backend flexibility too:
  - https://github.com/OWASP/Nest/blob/745539cd9af4ee3cd51ebbcbb33da2bd78347b34/backend/apps/owasp/api/search/chapter.py#L13
  - https://github.com/OWASP/Nest/blob/745539cd9af4ee3cd51ebbcbb33da2bd78347b34/backend/apps/owasp/api/search/project.py#L15
  
There should be a way to set params depending on the entity type (project, chapter, issue, etc)